src/substrait_extension.cpp

```diff
@@ -71,13 +71,26 @@ static void VerifySubstraitRoundtrip(unique_ptr<LogicalOperator> &query_plan, Co
                                      ToSubstraitFunctionData &data, const string &serialized, bool json) {
 	// We round-trip the generated json and verify if the result is the same
 	auto actual_result = con.Query(data.query);
+
 	auto sub_relation = SubstraitPlanToDuckDBRel(con, serialized, json);
 	auto substrait_result = sub_relation->Execute();
 	substrait_result->names = actual_result->names;
-	if (!actual_result->Equals(*substrait_result)) {
+	unique_ptr<MaterializedQueryResult> substrait_materialized;
+
+	if (substrait_result->type == QueryResultType::STREAM_RESULT) {
+		auto &stream_query = substrait_result->Cast<duckdb::StreamQueryResult>();
+
+		substrait_materialized = stream_query.Materialize();
+	} else if (substrait_result->type == QueryResultType::MATERIALIZED_RESULT) {
+		substrait_materialized = unique_ptr_cast<QueryResult, MaterializedQueryResult>(std::move(substrait_result));
+	}
```

Comment:
I was gonna say `else { throw InternalException }` but that will happen when we use `substrait_materialized` and it's nullptr anyways